### ISSUE TITLE
[CCRenderTexture newCGImage] error in 3.1

### ISSUE THREAD

**Issue by slembcke** [cocos2d/cocos2d-objc#743]
http://forum.cocos2d-iphone.org/t/how-to-do-gl-calls-with-the-new-render-queue/13340

**Comment by jpswensen** [comment]
I am not sure if this helps, but I debugged the OpenGL ES frames while running on my device and found something that may be strange. I noticed that on every single other operation in the queue that the operations were on the Renderbuffer #1, Draw framebuffer #2, and Read framebuffer #2. However, the glReadPixels had Draw framebuffer #3 and Read framebuffer #3. I have attached a few screenshots to show (1) when the fragment shader drew the HSV circle, (2) the CCRenderTexture begin/end, and (3) the HSVColorPicker: readPixels with the messed up framebuffer number.

![screen shot 2014-05-08 at 5-8 1 35 55 pm](https://cloud.githubusercontent.com/assets/722212/2919215/c225da1c-d6d8-11e3-976e-c1384961eeb2.png)
![screen shot 2014-05-08 at 5-8 1 37 18 pm](https://cloud.githubusercontent.com/assets/722212/2919214/c224f5e8-d6d8-11e3-8020-d9be0fbb8963.png)
![screen shot 2014-05-08 at 5-8 1 37 22 pm](https://cloud.githubusercontent.com/assets/722212/2919213/c21fd450-d6d8-11e3-9247-f518913b4810.png)
